rtsp.go

```diff
@@ -675,26 +669,33 @@ func newRTSPCamera(ctx context.Context, _ resource.Dependencies, conf resource.C
 	}
 	cancelCtx, cancel := context.WithCancel(context.Background())
 	reader := gostream.VideoReaderFunc(func(_ context.Context) (image.Image, func(), error) {
-		latest := rc.latestOutput.Load()
-		if latest == nil {
-			return nil, func() {}, errors.New("no frame yet")
+		if videoCodec(rc.currentCodec.Load()) == MJPEG {
+			img := rc.latestMJPEGImage.Load()
+			if img == nil {
+				return nil, func() {}, errors.New("no mjpeg frame yet")
+			}
+			return *img, func() {}, nil
 		}
 
-		release := func() {}
-		// When model is mjpeg, latest frame wrapper will be nil.
-		if latest.frameWrapper != nil {
-			latest.frameWrapper.isBeingServed.Store(true)
+		frame := rc.latestFrame.Load()
+		if frame == nil {
+			return nil, func() {}, errors.New("no frame yet")
+		}
 
-			release = func() {
-				// When the caller is done with the image, we return the AVFrame to the pool such
-				// that we can re-use its allocated byte buffer.
-				rc.logger.Debug("Release was called.")
-				latest.frameWrapper.isBeingServed.Store(false)
-				rc.avFramePool.safelyPut(latest.frameWrapper)
-			}
+		release := func() {
+			// When the caller is done with the image, we return the AVFrame to the pool such
+			// that we can re-use its allocated byte buffer.
+			rc.logger.Debug("Release was called.")
+			frame.isBeingServed.Store(false)
+			frame.refs.Add(-1)
+			rc.avFramePool.put(frame)
 		}
 
-		return latest.img, release, nil
+		defer func() {
```

Comment:
Can we justify the defer as a comment?

Reply:
Sure does

`// Handle refs and flags after we pass the frame on to the server`

work?

Reply:
The defer is still executed before the return value is propagated to the server. So that wouldn't be accurate. Put another way, what's the consequence of not doing this in a defer?

Reply:
Damn I misunderstood Golang. I thought deferred statements ran after when it really just guarantees The defer statement in Go essentially guarantees that certain logic will run at the very end of the function body, just before the function actually returns to its caller?

> what's the consequence of not doing this in a defer

Not much as long as we keep the lines right before the return. I guess the main difference and potential benefit would be that it is ran after all other deferred statements.

Would a better comment be 

`// Handle refs and flags right before handoff to server`

Reply:
Action taken: removed those two lines from defer